### PR TITLE
Add NoCacheWhenBuilding setting

### PR DESCRIPTION
**Pull request by Chemaclass**
### 🤔 Background

Fixes: https://github.com/phel-lang/phel-lang/issues/670

When building a phel project, you can enable/disable the cache for the already generated files. By default, the cache is enabled. This will result into loading using `require_once $target` which might endup into a weird behaviour if you got some files already built and you want to rebuild the project using the cache.

### 💡 Goal

Fix the build behaviour by allowing defining paths to `no cache` even when running build with cache.

### 🔖 Changes

- Added a new phel config setting: `PhelConfig->setNoCacheWhenBuilding(['path1', 'path2...'])`

### CODE REVIEW

**Review by JesusValera — APPROVED** [review]

**Comment by jenshaase** [comment]
@Chemaclass As far as I understand this is just a workaround and not causing the root of the problem, since the root of the problem is currently unknown?

**Comment by Chemaclass** [comment]
@jenshaase this is a workaround to the actual issue which I tried to explain here https://github.com/phel-lang/phel-lang/issues/670#issuecomment-2059470128

**Comment by jenshaase** [comment]
@Chemaclass Ok, the I approved it

**Review by jenshaase — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ All notable changes to this project will be documented in this file.
 * Add `phel fmt` alias for format (#673)
 * Add support for numeric on `empty?` (#683)
 * Fix `interleave` allowing nil keys and values (#682)
+* Add `phel fmt` alias for format
+* Add support for numeric on `empty?`
```
Comment (kambo-1st) [inline CHANGELOG.md:14]: Both of these lines seem like duplications of previous lines (10, 11).

Comment (Chemaclass) [reply CHANGELOG.md:14]: upsi! the git conflict... thanks!

**Comment by Chemaclass** [comment]
It's finally green! 🥳 
Screenshot 2024-04-16 at 22 12 31

**Pull request merged**